internal/client.go

```diff
@@ -721,7 +723,7 @@ type (
 
 		// StartDelay - Time to wait before dispatching the first workflow task.
 		// A signal from signal with start will not trigger a workflow task.
-		// Cannot be set the same time as a CronSchedule.
+		// Cannot be set the same time as a CronSchedule or WithStartOperation.
```

Comment:
Do we have a runtime check for these?

Reply:
The server implements these checks: see https://github.com/temporalio/temporal/blob/main/service/frontend/workflow_handler.go#L603-L614

I could also add a client-side check? I think we had discussed client-side vs server-side in a previous PR and couldn't quite find a good rule of thumb for when to use which.

Reply:
Yeah having client side validation is nice to have, but I think it is well out of scope of this PR